nifi-provenance-reporting-tasks/src/main/java/io/egm/nifi/reporting/EmailProvenanceReporter.java

```diff
@@ -212,24 +230,35 @@ public PasswordAuthentication getPasswordAuthentication() {
      * @return an InternetAddress[] parsed from the supplied property
      * @throws AddressException if the property cannot be parsed to a valid InternetAddress[]
      */
-    private InternetAddress[] toInetAddresses(final ReportingContext context, PropertyDescriptor propertyDescriptor)
+    private InternetAddress[] toInetAddresses(final ReportingContext context, final Map<String, Object> event, PropertyDescriptor propertyDescriptor)
             throws AddressException {
         InternetAddress[] parse;
-        final String value = context.getProperty(propertyDescriptor).getValue();
-        if (value == null || value.isEmpty()) {
-            if (propertyDescriptor.isRequired()) {
-                final String exceptionMsg = "Required property '" + propertyDescriptor.getDisplayName() + "' evaluates to an empty string.";
-                throw new AddressException(exceptionMsg);
-            } else {
-                parse = new InternetAddress[0];
-            }
-        } else {
+        String value;
+        if(propertyDescriptor == SPECIFIC_RECIPIENT_ATTRIBUTE_NAME) {
+            value = getSpecificRecipientValue(context, event);
```

Comment:
what if you don't find the attribute in the event?

Reply:
what is the best way to handle that case ? can I throw a `NullPointerException` in the `getSpecificRecipientValue ` method  ?

Reply:
you should never ever throw a `NullPointerException`.

having the attribute is optional so it should never raise an error. we have it: good. we don't have it: let's move on.